generators/servicehook/index.js

```diff
@@ -0,0 +1,258 @@
+﻿'use strict';
+const path = require('path');
+const util = require(`../app/utility`);
+var generators = require('yeoman-generator');
+var chalk = require('chalk');
+var yosay = require('yosay');
+var uuid = require('uuid');
+var ejs = require('ejs');
+
+function construct() {
+      // Calling the super constructor is important so our generator is correctly set up
+      generators.Base.apply(this, arguments);
+
+      this.argument('extName', { type: String, required: false, desc: 'Name of the extension' });
+}
+
+function input() {
+
+      // Collect any missing data from the user.
+      // This gives me access to the generator in the
+      // when callbacks of prompt
+      let cmdLnInput = this;
+
+      return this.prompt([
```

Comment:
FUTURE: should pull this common stuff (that's common across all generators) into a single class so you don't have to repeat the same input definitions over and over (and it's easier to change in one place later).